src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -331,28 +332,41 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
                 switch (elements[0])
                 {
                     case "Pid":
-                        results.Pid = int.Parse(elements[1]);
+                        int.TryParse(elements[1], out results.Pid);
                         break;
+
                     case "VmHWM":
-                        results.VmHWM = ulong.Parse(elements[1]) * 1024;
+                        ulong.TryParse(elements[1], out results.VmHWM);
+                        results.VmHWM *= 1024;
                         break;
+
                     case "VmRSS":
-                        results.VmRSS = ulong.Parse(elements[1]) * 1024;
+                        ulong.TryParse(elements[1], out results.VmRSS);
+                        results.VmRSS *= 1024;
```

Comment:
We're multiplying by 1024 in all of these cases, without paying attention to the unit used in the status file.  Is that unit guaranteed to be "kB" as it is in your example above?  Or is it possible it could be reported in bytes or megabytes or another unit?  If so, we'd need to parse that unit as well and multiply by the appropriate factor.

Reply:
Yes, the unit is guaranteed to be in kB always.

Reply:
can we add extra Debug.Assert with the check?

Reply:
> can we add extra Debug.Assert with the check?

It's very less likely that this will change. It has been hardcoded in the linux source code.

Ref: https://github.com/torvalds/linux/blob/master/fs/proc/task_mmu.c#L59-L77

If you still want, I can add one.